src/common/themeSelectors/buttonSizes/buttonSizes.test.ts

```diff
@@ -0,0 +1,35 @@
+import theme from '../theme/mock-theme.json';
+import * as selectors from './buttonSizes';
+
+const scenarios = [
+  {
+    expectedResult: theme.buttonSizes.medium,
+    name: 'getButtonPropsBySize',
+    params: 'medium',
+    selector: selectors.getButtonPropsBySize,
+    title: 'button',
+  },
+  {
+    expectedResult: theme.radius.medium,
+    name: 'getRadiusBySize',
+    params: 'medium',
+    selector: selectors.getRadiusBySize,
+    title: 'radius',
+  },
+];
+
+describe('Button Sizes selectors', () => {
+  /*  eslint-disable mocha/no-setup-in-describe */
+  scenarios.forEach(scenario => (
```

Comment:
Entendo que foi criado um `forEach` para rodar 2 cenários reaproveitando código. Porém, idealmente, o código de testes deve evitar muitas abstrações, porque ele é a documentação do código de produção.

Ainda mais que, pelo que entendi, são apenas 2 cenários. Se fosse um número crescente e indeterminado de cenários, valeria a pena uma abstração. Assim, eu sugeriria criar um `describe` para o cenário de `button` e outro `describe` para o cenário `radius`. O arquivo provavelmente continuará pequeno.

Porém, se essa lista de cenários, na verdade, for crescer muito, eu tenho outra sugestão:

Para mitigar o problema de `no-setup-in-describe`, ter mais aderência à API do jest e evitar _performance issues_, idealmente poderíamos usar `describe.each` aqui. Ainda mais por estarmos abordado test scenarios.

Ficaria* assim:

```ts
describe.each(scenarios)("when in scenario %#", (scenario) => {
    it(`should return the ${scenario.title} sizes`, () => {
      const result = scenario.selector(theme, scenario.params as selectors.Size);

      expect(result).toBe(scenario.expectedResult);
})
```

De toda forma, eu recomendaria evitar ao máximo essa abordagem, porque ela não descreve o que estou testando.

Reply:
eu super defendo scenarios nesse caso. esses testes são testes de seletores, funções que tem muito pouca lógica e os testes de todos os seletores são exatamente iguais: chama o seletor, compara com o output.
Esse arquivo no momento tem somente 2 seletores, mas se tu olhar cores por ex, são muitos!
quando a gente trocar pra fonte da verdade, esse contrato vai mudar, imagina manter 8 arquivos de testes de seletores cada um com N seletores e ter que atualizar isso um a um?
acho que teu ponto sobre abstração de testes muito válido! mas acho que manutenção deles também deve ser levada em consideração.
Se um dos testes dos seletores falhar, ele te diz exatamente qual e porque falhou, e nosso `it` tem apenas uma linha e eu acho que essa linha descreve bem o que esta sendo testado.

Reply:
Curti a sugestão do `describe.each`!

Reply:
@arielwb faz sentido. Inclusive, eu notei que, em um outro trecho de código, tem um caso com muitos cenários mesmo: https://github.com/natura-cosmeticos/natds-rn/pull/71#discussion_r474018427

Ali realmente, faz muito sentido termos essa abstração. Com minha sugestão, só trocaríamos o `forEach` por `describe.each` e se um dos seletores falhar, ainda é possível saber em qual cenário a falha aconteceu.